src/Core/Algorithms/Legacy/Inverse/TikhonovAlgoAbstractBase.cc

```diff
@@ -0,0 +1,503 @@
+				return false;
+            }
+        }
+    }
+
+    // check measurement regularization matrix sizes
+    if (sensorWeighting_)
+    {
+        if (get(Parameters::regularizationResidualSubcase).toInt() == residual_constrained)
+        {
+            // check that the matrix is of appropriate size (equal number of rows as rows in fwd matrix)
+            if(M != sensorWeighting_->ncols())
+            {
+				THROW_ALGORITHM_INPUT_ERROR("Data Residual Weighting Matrix must have the same number of rows as the Forward Matrix !");
+				return false;
+            }
+        }
+        // otherwise if the source covariance matrix is provided in squared form
+        else if  ( get(Parameters::regularizationResidualSubcase).toInt() == residual_constrained_squared )
+        {
+            // check that the matrix is of appropriate size and squared (equal number of rows as rows in fwd matrix)
+            if( (M != sensorWeighting_->nrows()) || (M != sensorWeighting_->ncols()) )
+            {
+				THROW_ALGORITHM_INPUT_ERROR("Squared data Residual Weighting Matrix must have the same number of rows and columns as number of rows in the Forward Matrix !");
+				return false;
+			}
+
+        }
+    }
+
+    return true;
+
+}
+
+
+/////////////////////////
+/////////  run()
+AlgorithmOutput TikhonovAlgoAbstractBase::run(const AlgorithmInput & input) const
+{
+	// get inputs
+	auto forwardMatrix_ = input.get<Matrix>(TikhonovAlgoAbstractBase::ForwardMatrix);
+	auto measuredData_ = input.get<Matrix>(TikhonovAlgoAbstractBase::MeasuredPotentials);
+	auto sourceWeighting_ = input.get<Matrix>(TikhonovAlgoAbstractBase::WeightingInSourceSpace);
+	auto sensorWeighting_ = input.get<Matrix>(TikhonovAlgoAbstractBase::WeightingInSensorSpace);
+
+	// get Parameters
+	auto RegularizationMethod_gotten = getOption(Parameters::RegularizationMethod);
+	auto TikhonovImplementation_gotten = get(Parameters::TikhonovImplementation).toString();
+
+    // Alocate Variable
+	DenseMatrix solution;
+    double lambda = 0;
+	double lambda_ = 0;
+
+	// check input MATRICES
+	checkInputMatrixSizes( input );
+
+	// Determine specific Tikhonov Implementation
+	TikhonovImpl  *algoImpl;
+	if ( TikhonovImplementation_gotten ==  std::string("standardTikhonov") ){
+
+		// get Parameters
+		int  regularizationChoice_ = get(Parameters::regularizationChoice).toInt();
+		int regularizationSolutionSubcase_ = get(Parameters::regularizationSolutionSubcase).toInt();
+		int regularizationResidualSubcase_ = get(Parameters::regularizationResidualSubcase).toInt();
+
+		algoImpl = new SolveInverseProblemWithStandardTikhonovImpl( *castMatrix::toDense(forwardMatrix_), *castMatrix::toDense(measuredData_), *castMatrix::toDense(sourceWeighting_), *castMatrix::toDense(sensorWeighting_), regularizationChoice_, regularizationSolutionSubcase_, regularizationResidualSubcase_);
```

Comment:
@jcollfont @jessdtate I missed reviewing this file since it was collapsed in the larger diff view. This is the one with multiple memory leak problems. Can someone fix it soon? It may have to be me...